src/JitInterface/src/ThunkGenerator/Program.cs

```diff
@@ -272,6 +273,88 @@ unsafe partial class CorInfoImpl
             }
             tr.WriteLine();
 
+            foreach (FunctionDecl decl in functionData)
+            {
+                string returnType = decl.ReturnType.ManagedTypeName;
+                int marshalAs = returnType.LastIndexOf(']');
```

Comment:
Why is this called "marshalAs"? Is this checking presence of MarshalAs attribute associated with return type? If so, will this handle return type that are arrays (e.g. string[])?